### ISSUE TITLE
 wpf Fontawesome icons are not showing at runtime

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/wpf#4923]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/wpf-Fontawesome-icons-are-not-showing-at/1483207)._

---
[severity:I'm unable to use this version]
I am writing a WPF application and I'm using fontawesome icons as part of my UI.

At design time everything renders fine, as seen in the first image, which basically means the fonts are resolved fine.

But when I run the program the icons show up as empty squares as shown in the second image.

These are the Steps I took to get to the issue.

1. I downloaded the fontawesome.ttf file
2. In Visual studio I created a new WPF project. (.NET Core 5)
3. I created a new folder called fonts and placed the .ttf file in the same folder
4. I set the property of the .ttf file as a resource. (as for the other properties I have tried all the iterations like "copy always" or "copy when newer"
5. I rebuilt the project.
6. Then I added the ttf font as a resource in the proper way 

<FontFamily x:Key="FontAwesomeFont" >pack://application:,,,/Fonts/#Font Awesome 5 Free Solid</FontFamily>

(I have also tried multiple variations of this even though this is the recommended  way according to the documentation.)
7. Then I used the font as usual.

When I use the regular characters ie. A - Z, 1 - 9 things work fine both at design time and runtime.
But when I use the icons ie characters with "&$#f0110;" for example, the icons show up fine at design time but when I run the program it shows squares.

Other things I have tried are

Cleaning solution and recompiling.
Starting a brand new solution and only test the font
Cleaning the solution, Deleting all the bin and obj folders and recompiling and running the solution again.

And many more things but these are the most important ones.

PLEASE HELP. I do think there may be a bug.
Other youtubers seem to be having no issues whatsoever and I followed their work step by step.

Thanks for any help

---
#### Feedback Bot on 7/22/2021, 10:38 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by Metritutus** [comment]
From the information supplied, that icon reference doesn't seem correct.  The project won't even build if you use that character sequence (`&$#f0110;`).

First, for the prefix, you need to remove the `$` and add `x`, making the prefix `&#x`.  This is the prefix for an XML hex character reference.

Second, you want to remove the first `0` from the hex unicode identifier as `f0110` does not appear to the unicode reference of an actual Font Awesome character.  This will make the identifier `f110` (which is apparently the [spinner icon](https://fontawesome.com/icons/spinner)).

The full character reference will end up being `&#xf110;`, which should work as expected provided the Font Awesome OTF files have been referenced correctly.

With the Font Awesome OTF files placed in a folder called `Fonts`, I tested this successfully with the following XAML:
```xaml
<Window x:Class="FontAwesomeTest001.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d"
        Title="MainWindow" Height="450" Width="800">
    <Window.Resources>
        <ResourceDictionary>
            <FontFamily x:Key="FontAwesome">pack://application:,,,/Fonts/#Font Awesome 5 Free Solid</FontFamily>
        </ResourceDictionary>
    </Window.Resources>
    <Grid>
        <TextBlock Text="&#xf110;" FontFamily="{StaticResource FontAwesome}"></TextBlock>
    </Grid>
</Window>
```

Also, in testing the above and experimenting with different (invalid) ways of referencing the Font Awesome character, the Designer appeared to behave as expected, showing a square when it couldn't find the character, or displaying `Invalid Markup` when using a character sequence that would prevent the project from building (such as the character sequence specified in the original post).

**Comment by MichaelWatindi** [comment]
> From the information supplied, that icon reference doesn't seem correct. The project won't even build if you use that character sequence (`&$#f0110;`).
> 
> First, for the prefix, you need to remove the `$` and add `x`, making the prefix `&#x`. This is the prefix for an XML hex character reference.
> 
> Second, you want to remove the first `0` from the hex unicode identifier as `f0110` does not appear to the unicode reference of an actual Font Awesome character. This will make the identifier `f110` (which is apparently the [spinner icon](https://fontawesome.com/icons/spinner)).
> 
> The full character reference will end up being `&#xf110;`, which should work as expected provided the Font Awesome OTF files have been referenced correctly.
> 
> With the Font Awesome OTF files placed in a folder called `Fonts`, I tested this successfully with the following XAML:
> 
> ```
> <Window x:Class="FontAwesomeTest001.MainWindow"
>         xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
>         xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
>         xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
>         xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
>         mc:Ignorable="d"
>         Title="MainWindow" Height="450" Width="800">
>     <Window.Resources>
>         <ResourceDictionary>
>             <FontFamily x:Key="FontAwesome">pack://application:,,,/Fonts/#Font Awesome 5 Free Solid</FontFamily>
>         </ResourceDictionary>
>     </Window.Resources>
>     <Grid>
>         <TextBlock Text="&#xf110;" FontFamily="{StaticResource FontAwesome}"></TextBlock>
>     </Grid>
> </Window>
> ```
> 
> Also, in testing the above and experimenting with different (invalid) ways of referencing the Font Awesome character, the Designer appeared to behave as expected, showing a square when it couldn't find the character, or displaying `Invalid Markup` when using a character sequence that would prevent the project from building (such as the character sequence specified in the original post).

Hi Metritutus, 
Thank you for the quick response. The question was originally my post. I must've mistyped the prefix. I am using the exact same one you are using in your code, i.e. "&#xf110".
Your solution helped greatly. On a fresh project, I was able to get the icons to show at runtime. However not in my larger project which has multiple projects that are referencing each other. I believe I'm not referencing the fonts correctly. 
I am puzzled as to why the icons would show at design time though, but not when I run it. 
Thanks again

**Comment by MichaelWatindi** [comment]
I was able to figure it out. When the solution has multiple projects we need to use the project name plus the component keyword in the fontfamily definition. For example:
<FontFamily x:Key="FontAwesome">pack://application:,,,/<PROJECT_NAME>;COMPONENTS;/Fonts/#Font Awesome 5 Free Solid</FontFamily>
I had the application running on one project and the controls, styles and fonts in another project. I thought since they fonts and windows, styles, controls were all in a single project I did not need to put the project name. But after I did this it worked.
Thanks for the response.
